### PR TITLE
Fix (bsc#1180721) libgpgme started to return signatures of unknown keys

### PR DESCRIPTION
**Pull request by bzeller**

### CODE REVIEW

File: zypp/KeyManager.cc [hunk 1]
```diff
@@ -182,24 +182,33 @@ std::list<std::string> KeyManagerCtx::Impl::readSignaturesFprsOptVerify( const P
...
-      // bsc#1100427: With ibgpgme11-1.11.0 and if a recent gpg version was used
+      // bsc#1100427: With libgpgme11-1.11.0 and if a recent gpg version was used
       // to create the signature, the field may contain the full fingerprint, but
       // we're expected to return the ID.
       // [https://github.com/gpg/gpgme/commit/478d1650bbef84958ccce439fac982ef57b16cd0]
       std::string id( sig->fpr );
       if ( id.size() > 16 )
 	id = id.substr( id.size()-16 );
+
+      DBG << "Found signature with ID: " << id  << std::endl;
```
Comment (mlandres) [inline zypp/KeyManager.cc:199]:
```suggestion
      DBG << "Found signature with ID: " << id  << " in " << file_r << std::endl;
```

File: zypp/KeyManager.cc [hunk 2]
```diff
@@ -182,24 +182,33 @@ std::list<std::string> KeyManagerCtx::Impl::readSignaturesFprsOptVerify( const P
...
       signatures.push_back( std::move(id) );
     }
 
-    if ( sig->status != GPG_ERR_NO_ERROR )
-    {
-      if ( gpgme_err_code(sig->status) != GPG_ERR_KEY_EXPIRED )
+    if ( sig->status != GPG_ERR_NO_ERROR ) {
+      DBG << "Verification failed with result: " << gpgme_err_code(sig->status) << std::endl;
```
Comment (mlandres) [inline zypp/KeyManager.cc:204]:
```suggestion
      DBG << "Verification failed with result: " << file_r << " " << gpgme_err_code(sig->status) << std::endl;
```

**Review by mlandres — CHANGES_REQUESTED** [review]
Just some debug message enhancement.

**Review by mlandres — APPROVED** [review]

File: zypp/KeyManager.cc (hunk 1 above)
Comment (bzeller) [reply zypp/KeyManager.cc]: OK

File: zypp/KeyManager.cc (hunk 2 above)
Comment (bzeller) [reply zypp/KeyManager.cc]: Ok

**Pull request merged**